pkg/install/generate.go

```diff
@@ -615,3 +648,64 @@ func completeOpenStackDeprovisionJob(req *hivev1.ClusterDeprovision, job *batchv
 	job.Spec.Template.Spec.Containers = containers
 	job.Spec.Template.Spec.Volumes = volumes
 }
+
+func completeVSphereDeprovisionJob(req *hivev1.ClusterDeprovision, job *batchv1.Job) {
+	volumes := []corev1.Volume{}
+	volumeMounts := []corev1.VolumeMount{}
+	credentialsSecret := ""
+	if len(req.Spec.Platform.VSphere.CredentialsSecretRef.Name) > 0 {
+		credentialsSecret = req.Spec.Platform.VSphere.CredentialsSecretRef.Name
+	}
+	env := []corev1.EnvVar{}
+	if len(credentialsSecret) > 0 {
+		env = append(
+			env,
+			corev1.EnvVar{
+				Name: "GOVC_HOST",
+				ValueFrom: &corev1.EnvVarSource{
+					SecretKeyRef: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{Name: credentialsSecret},
+						Key:                  "govc_host",
+					},
+				},
+			},
+			corev1.EnvVar{
+				Name: "GOVC_USERNAME",
+				ValueFrom: &corev1.EnvVarSource{
+					SecretKeyRef: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{Name: credentialsSecret},
+						Key:                  "govc_username",
+					},
+				},
+			},
+			corev1.EnvVar{
+				Name: "GOVC_PASSWORD",
+				ValueFrom: &corev1.EnvVarSource{
+					SecretKeyRef: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{Name: credentialsSecret},
+						Key:                  "govc_password",
+					},
+				},
+			},
+		)
+	}
+	containers := []corev1.Container{
+		{
+			Name:            "deprovision",
+			Image:           images.GetHiveImage(),
+			ImagePullPolicy: images.GetHiveImagePullPolicy(),
+			Env:             env,
+			Command:         []string{"/usr/bin/hiveutil"},
+			Args: []string{
+				"deprovision",
+				"vsphere",
+				"--loglevel",
+				"debug",
+				req.Spec.InfraID,
+			},
```

Comment:
There needs to be a `--vcenter` flag. Or the `hiveutil deprovision vsphere` command needs to use GOVC_HOST env var instead.